### ISSUE TITLE
Docs: Missing the need for java 8

### ISSUE THREAD

**Issue by ColtonIdle** [facebook/flipper#512]
Setting up flipper on a brand new project fails due to a gap in the docs. You need to explicitly target java 8 in Android.

You'll get a crash similar to:

```
2019-08-09 14:03:13.433 24932-24932/com.example.sample E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.sample, PID: 24932
    java.lang.BootstrapMethodError: Exception from call site #4 bootstrap method
        at okhttp3.internal.Util.<clinit>(Util.java:87)
        at okhttp3.internal.Util.immutableList(Util.java:234)
        at okhttp3.OkHttpClient.<clinit>(OkHttpClient.java:124)
        at okhttp3.OkHttpClient$Builder.<init>(OkHttpClient.java:449)
        at com.example.sample.MainActivity.onCreate(MainActivity.kt:46)
        at android.app.Activity.performCreate(Activity.java:7136)
        at android.app.Activity.performCreate(Activity.java:7127)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2893)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.ClassCastException: Bootstrap method returned null
```

Fix:

```
   compileOptions {
        targetCompatibility = "8"
        sourceCompatibility = "8"
    }
```

**Comment by passy** [comment]
I'm not sure we have a good place to inform users about this. This doesn't come from a Flipper requirement, but is because of the dependency on okhttp3. If you don't make use of it, the easiest alternative solution is to exclude the dependency.

**Comment by ColtonIdle** [comment]
Makes sense. But if all someone does is add the dependencies from the docs here, they are greeted with the error above. I think it's easier to just leave a note or warning saying that because of okhttp they will need to use java8 or exclude. I've setup Flipper like 3 different times across projects, and then I wasted an hour on this because I thought I actually messed up something else in my code while setting up a new project.
